lib/ts/recipe/emailverification/components/features/emailVerification/index.tsx

```diff
@@ -110,7 +114,7 @@ export const EmailVerification: React.FC<Prop> = (props) => {
 
     const signOut = useCallback(async (): Promise<void> => {
         const session = Session.getInstanceOrThrow();
-        await session.signOut(props.userContext);
+        await session.signOut({ userContext: props.userContext ?? userContext });
         return redirectToAuthWithHistory();
     }, [props.recipe, redirectToAuthWithHistory]);
```

Comment:
shouldn't the dependency array have userContext in it too?

Reply:
done